### PR TITLE
(kanban): create Kanban widget

### PR DESCRIPTION
**Pull request by rorychatt**
https://github.com/user-attachments/assets/4c667738-99b4-4c53-a1a4-b5a4e86c4ead

### CODE REVIEW

File: frontend/src/widgets/widgetMap.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import * as Blades from '@/widgets/blades';
 import * as Lists from '@/widgets/lists';
 import * as Details from '@/widgets/details';
 import * as Primitives from '@/widgets/primitives';
+import * as Kanban from '@/widgets/kanban';
```
Comment (dcrjodle) [inline frontend/src/widgets/widgetMap.ts]: Let's decide for uniformity what's the best import approach. `{item1, item2, item3} from 'module'` or `* as module from 'module'`

Comment (rorychatt) [reply frontend/src/widgets/widgetMap.ts]: Closing

As discussed, we will refactor all into the first style

File: frontend/src/widgets/index.ts [hunk 2]
```diff
@@ -16,3 +16,4 @@ export { SlotWidget } from './SlotWidget';
 export { TooltipWidget } from './TooltipWidget';
 
 export { PaginationWidget } from './PaginationWidget';
+export { KanbanWidget, KanbanColumnWidget, KanbanCardWidget } from './kanban';
```
Comment (nielsbosma) [inline frontend/src/widgets/index.ts:19]: Maybe we should make Kanban a Suspense widget?

File: Ivy.Samples.Shared/Apps/Widgets/KanbanApp.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        return Layout.Vertical(
+            Text.H3("Task Board Demo"),
+            Text.P("Showcasing kanban features: field selectors, column/card ordering, custom column titles, and event handlers."),
+
+            // Kanban with common features
+            tasks
+                .ToKanban(
+                    groupBySelector: e => e.Status,
+                    cardIdSelector: e => e.Id,
```
Comment (nielsbosma) [inline Ivy.Samples.Shared/Apps/Widgets/KanbanApp.cs]: rename to idSelector, titleSelector, descriptionSelector,

File: Ivy/Core/AbstractWidget.cs [hunk 4]
```diff
@@ -164,44 +164,67 @@ public bool InvokeEvent(string eventName, JsonArray args)
         if (eventDelegate == null)
             return false;
 
-        if (IsFunc(eventDelegate, out Type? eventType, out Type? returnType))
+        if (IsFunc(eventDelegate, out Type? eventType, out Type? returnType) && returnType == typeof(ValueTask))
```
Comment (nielsbosma) [inline Ivy/Core/AbstractWidget.cs:167]: Looks good, but this is starting to become quite a complicated mess. Think we need to refactor this and move most of this into Utils functions and create unit tests to makes sure we handle all corner cases.

**Review by nielsbosma — CHANGES_REQUESTED** [review]
How are these colors defined?

Image

File: Ivy.Samples.Shared/Apps/Widgets/KanbanApp.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                    cardTitleSelector: e => e.Title,
+                    cardDescriptionSelector: e => e.Description)
+                .ColumnOrder(e => e.StatusOrder)
+                .CardOrder(e => e.Priority)
```
Comment (rorychatt) [inline Ivy.Samples.Shared/Apps/Widgets/KanbanApp.cs]: use order selector

File: frontend/src/widgets/index.ts (hunk 2 above)
Comment (rorychatt) [reply frontend/src/widgets/index.ts:19]: Lazy load it in widget map -> will try out a heavy example and see if suspense is needed
